#!/usr/bin/env python
PACKAGE = "radar_experiments"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("yaw_tolerance",              double_t, 0, "Yaw tolerance window",                              0.04, 0,   0.3)
gen.add("ransac_distance_threshold",  double_t, 0, "Ransac threshold for line segmentation",            0.05, 0,   1)
gen.add("beta",                       double_t, 0, "Beta algorithm param",                              0.2,  0,   1)
gen.add("distance_threshold",         double_t, 0, "d_F param",                                         0.2,  0,   3)
gen.add("time_tolerance",             double_t, 0, "Time tolerance between lidar and radar timestamps", 0.3,  0,   1)
gen.add("radar_dev",                  double_t, 0, "Radar standard deviation",                          0.03, 0,   0.3)
gen.add("lidar_dev",                  double_t, 0, "Lidar standard deviation",                          0.02, 0,   0.3)
gen.add("ransac_iterations",          int_t, 0,    "Ransac iterations(number of lines to search for)",  6,    1,   20)
gen.add("min_ransac_pointcloud_size", int_t, 0,    "When to stop searching for lines",                  15,   2,   500)

exit(gen.generate(PACKAGE, "lidar_radar_fusion_node", "LidarFusion"))